econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -214,11 +215,26 @@ def _sanitize(obj):
             raise ValueError(f"Could not sanitize input {obj}")
 
 
+# Convert SingleTreeInterpreter to a python dictionary
+def _tree_interpreter_to_dict(interp, features, leaf_data=lambda t, n: {}):
+    tree = interp.tree_model_.tree_
+
+    def recurse(node_id):
+        if tree.children_left[node_id] == _tree.TREE_LEAF:
+            return {'leaf': True, 'n_samples': tree.n_node_samples[node_id], **leaf_data(tree, node_id)}
+        else:
+            return {'leaf': False, 'feature': features[tree.feature[node_id]], 'threshold': tree.threshold[node_id],
+                    'left': recurse(tree.children_left[node_id]),
+                    'right': recurse(tree.children_right[node_id])}
+
+    return recurse(0)
+
+
 # named tuple type for storing results inside CausalAnalysis class;
 # must be lifted to module level to enable pickling
 _result = namedtuple("_result", field_names=[
     "feature_index", "feature_name", "feature_baseline", "feature_levels", "hinds",
-    "X_transformer", "W_transformer", "estimator", "global_inference"])
+    "X_transformer", "W_transformer", "estimator", "global_inference", "treatment_value"])
 
 
 class CausalAnalysis:
```

Comment:
If the only thing that has changed in between runs for a particular feature are it's corresponding "heterogeneity_inds", then we should not repeat the model selection for model_t (if warm_start=True) as it is exactly the same call, since WX has not changed.

So we should add this logic in line 494.

Reply:
Also why are we retraining a model_y if self.heterogeneity_model has changed? The call to model_y model selection is exactly the same.

Also even if we want to go with that, then why are we not also re-training model_y if hetergogeneity_inds has changed but heterogeneity_model hasnt.

I think in either case retraining model_y is not needed.

Reply:
Also weirdly I got the warning:
`Using a linear model instead of a forest model for feature 'IsSupport' because forests don't support models with no heterogeneity indices`

Even when I specified `heterogeneity_model='linear'` and an empty list for hterogeneity_inds. This should only be raised if `hetergeneity='forest'`

Reply:
I have fixed the last issue with the warning.

At the moment the logic is relatively simple: we have a single "warm_start" flag which we choose not to honor if certain things have changed which would invalidate part of the already-computed results, and we have feature processing code that we run in its entirety for every feature that has been added/updated.  (But note that contra the second half of your second comment, we already *do not* retrain the y_model if only the heterogeneity indices have changed, we only ignore the warm_start flag if the nuisance models or  heterogeneity model has changed, since that invalidates all existing estimators)

It is possible to add additional checks to optimize out even more computation as you've suggested here, but I fear that it will be harder to verify the correctness of the resulting logic (e.g. how do you add a test that verifies that the y_model is trained only when necessary?).  How critical is optimizing the warm_start logic to eliminate all redundant computation, as opposed to having it be a convenience for adding a few more features?